proposals/lip-0048.md

```diff
@@ -78,43 +69,43 @@ The Fee module does not expose any functions.
 
 #### Verify Transaction
 
-During `trs` verification, the following logic is applied:
-
 ```python
-minFee = MIN_FEE_PER_BYTE * size(trs)
-         + extraCommandFee(trs.moduleID, trs.commandID)
-if trs.fee < minFee:
-     trs is invalid
+def verify(trs: Transaction) -> None:
+    trsSize = length(encode(transactionSchema), trs)
+    minFee = MIN_FEE_PER_BYTE * trsSize
+    if trs.fee < minFee:
+         raise Exception("Insufficient transaction fee")
+    senderAddress = SHA256(trs.senderPublicKey)[:LENGTH_ADDRESS]
+    balance = Token.getAvailableBalance(senderAddress, TOKEN_ID_FEE)
+    if trs.fee > balance:
+        raise Exception("Insufficient balance")
 ```
 
-#### Before Transaction Execution
-
-Before a transaction `trs` is executed, the following logic is applied:
+#### Before Command Execution
 
 ```python
-minFee = MIN_FEE_PER_BYTE * size(trs)
-         + extraCommandFee(trs.moduleID, trs.commandID)
-senderAddress is derived from trs.senderPublicKey
-generatorAddress is the address of the generator of the block including trs
-
-if TOKEN_ID_FEE.chainID == 0: # Fee token is a native token
-    token.burn(senderAddress,
-               TOKEN_ID_FEE,
-               minFee)
-    token.transfer(senderAddress,
-                   generatorAddress,
+def beforeCommandExecute(trs: Transaction, b: Block) -> None:
+    trsSize = length(encode(transactionSchema), trs)
```

Comment:
```suggestion
    trsSize = len(encodeTransaction trs)
```

Reply:
encode function in transaction schema LIP was renamed..

Reply:
shouldnt it be `encodeTransaction(trs)`

Reply:
Fixed.